feat: API 요청 key를 번갈아 사용하도록 설정한다

kiarakim opened the pull request:
## 📄 Summary
> 

## 🙋🏻 More
>

kiarakim committed:
refactor: 도커 로그인 정보 명시

kiarakim committed:
refactor: 메서드 명 변경

sosow0212 committed:
feat:  상세 정보 조회 API를 작성한다 (#53)

* refactor: Test Fixture에 누락된 데이터 추가

[#23]

* feat: 충전소를 id 값 기준으로 상세 조회하는 기능 구현

[#23]

* docs: 충전소를 id 값 기준으로 상세 조회하는 기능 Rest Docs 문서 작성

[#23]

* refactor: 패키지 구조 변경

[#23]

* test: repository 테스트 추가

[#23]

* refactor: 팀 컨벤션에 따라 변수명 변경

[#23]

* refactor: @Transactional 어노테이션 추가

[#23]

* refactor: 개행 및 final 키워드 컨벤션에 맞게 제거

[#23]

* refactor: 예외 메시지 검증

[#23]

* refactor: 공공 API에서 주소도 받을 수 있도록 리팩토링 진행

[#23]

kyw0716 committed:
feat: msw로 충전소 데이터를 mocking하는 기능을 구현한다 (#56)

* refactor: Position 타입 → google.maps.LatLngLiteral로 변경

- 사용자 정의 타입에서 구글이 기본으로 제공해주는 타입으로 변경

[#47]

* feat: msw로 충전소 데이터를 mocking하는 기능 구현

[#47]

Co-authored-by: Gabriel Ju Hyun, Yoon <gabrielyoon7@gmail.com>
Co-authored-by: Dain Lee <feb.dain@gmail.com>

kyw0716 committed:
feat: TanStack Query를 사용하여 충전소 마커를 찍는다 (#57)

* chore: @tanstack/react-query @tanstack/react-query-devtools 패키지 설치 및 세팅

- 초기 환경 설정
  - Provider 추가
- @tanstack/react-query 설치
- @tanstack/react-query-devtools 설치

[#52]

* fix: Station 타입 변경 사항 반영

[#52]

* chore: StrictMode 해제

- 의도하지 않은 중복 렌더링, 이벤트 등록을 피하기 위해 StrictMode 해제

[#52]

* feat: TanStack Query를 사용하여 충전소 마커를 찍는 기능 구현

[#52]

---

Co-authored-by: Gabriel Ju Hyun, Yoon <gabrielyoon7@gmail.com>
Co-authored-by: Dain Lee <feb.dain@gmail.com>

be-student committed:
feat: CODEOWNERS 파일 작성 (#55)

[#8]

feb-dain committed:
style: prettier 적용 안 된 파일에 prettier를 적용한다  (#63)

* style: prettier 적용

[#62]

* rename: tsx → ts로 확장자 변경

[#62]

be-student committed:
feat: logger error 마다 슬랙으로 발생하는 로그를 전송한다 (#54)

* feat: logger error 마다 슬랙으로 발생하는 로그를 전송한다

[#9]

* fix: webhook 부분을 환경 변수로 가져오도록 변경

[#9]

* style: 개행 추가 및, final 키워드 제거

[#9]

* fix: local 이 아닐때, 보내도록 수정

[#9]

* fix: error 레벨일때만 보내도록 변경

[#9]

kiarakim committed:
rebase

kiarakim committed:
feat: API 요청 key를 번갈아 사용하는 기능 추가

be-student commented on backend/src/main/java/com/carffeine/carffeine/service/ScrapperService.java:
```diff
@@ -27,11 +28,17 @@ public class ScrapperService {
     private final RestTemplate restTemplate;
     private final ChargeStationRepository chargeStationRepository;
+
+    private final Random random = new Random();
```
```suggestion
     private final ThreadLocalRandom threadLocalRandom= ThreadLocalRandom.current();
```

참고
https://velog.io/@sojukang/Random-%EB%8C%80%EC%8B%A0-ThreadLocalRandom%EC%9D%84-%EC%8D%A8%EC%95%BC-%ED%95%98%EB%8A%94-%EC%9D%B4%EC%9C%A0

kiarakim replied:
Random이 완전한 랜덤이 아니었다니 신기하네요.
새로운 기능 감사합니다~

be-student commented on backend/src/main/resources/application.yml:
```diff
@@ -2,10 +2,12 @@ server:
-    url: ENC(DR0RaV+v+UeGBHYRkdmVN7zqR782LpqPp+83nh7Jd2Wl/f13pCNHcORYANVXVNwumhEIYlZhJT9jVNpI3y0AbJ+S6wJ8JX6zyRuseCh9E9ux1fn1HvuxZhO2nbR3h8OEpa1l7s2G1doZHXjKVwrZnxdxN+uwJ3kCAskv7VOvRSA=)
-    username: ENC(Pcbw8DunWO7Gw8Ji1aZZDw==)
-    password: ENC(cC5dJEszKWMzZLKIgNs14/PEqCtjA5nP)
+    url: ENC(D6dCO3e8DX1Wuj6p5ZWMm9Cfk/k6YKeTa9qkVMqe4jEl9/Euw8AEcMBHgntlaZYGlMnw0sEvtXSu1ZyHLmQioxRS3Kdjd5OKczqSVA72eI9qldasSyoSLEDKBkBly8MYdsKOPpGIH08S9IG6QculCz66TDuBNNZEldt8GcQHAeM=)
+    username: root
+    password: ENC(0OkM1cOAS6M6p3TO8BJDMU3eBtGcB0/E)
```
이 부분도 localhost 로 직접 하드코딩해도 좋을 것 같아요

kiarakim replied:
이부분을 하드코딩해도 되는지 몰랐네요. 감사합니다!

be-student reviewed (changes requested):
gitignore 에 mysql 폴더를 추가하면 좋을 것 같아요

서비스 로직에, key 를 관리하는 코드를 넣는 것보다는 랜덤하게 key 를 반환해주는 컴포넌트를 추가하면 어떨까요?

gabrielyoon7 committed:
feat: google maps api와 react UI를 연동한다 (#68)

* refactor: 폴더 구조 개선

- google maps api 관련 컴포넌트 이동

Co-authored-by: Dain Lee <feb-dain@users.noreply.github.com>
Co-authored-by: YoungWoo Kim <kyw0716@users.noreply.github.com>

[#61]

* feat: 마커 클릭시 충전소 간단 정보 컴포넌트 뜨는 기능 구현

Co-authored-by: Dain Lee <feb-dain@users.noreply.github.com>
Co-authored-by: YoungWoo Kim <kyw0716@users.noreply.github.com>

[#61]

Co-authored-by: kyw0716 <kyw0716@naver.com>
Co-authored-by: Dain Lee <feb.dain@gmail.com>

kiarakim committed:
rebase

kiarakim committed:
refactor: 난수 발생 로직 변경

gabrielyoon7 committed:
refactor: msw 및 데이터를 개선한다 (#72)

* refactor: msw의 불필요한 경고 메세지 제거

- [문서 참고](https://mswjs.io/docs/api/setup-worker/start#onunhandledrequest)

* refactor: 충전소 랜덤 mock 데이터가 좀 더 현실적으로 생성되도록 기능 개선

* style: prettier 적용

[#71]

feb-dain committed:
refactor: 코드 가독성을 증진시킨다 (#74)

* refactor: 파라미터명 명확화

- map → googleMap

[#70]

* refactor: googleMap 타입 전역으로 사용

[#70]

* refactor: 필요 없는 as(타입 단언) 제거

[#70]

* refactor: 타입 중복 제거

[#70]

* refactor: 오타 수정

[#70]

* refactor: googleMap 타입 전역 사용 취소

[#70]

* refactor: useMutation 훅 분리

[#70]

* refactor: CarFfeineMap 컴포넌트 아래에 CarFfeineMapListener 배치

[#70]

-

Co-authored-by: Gabriel Ju Hyun, Yoon <gabrielyoon7@gmail.com>
Co-authored-by: kyw0716 <kyw0716@naver.com>

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/domain/RandomKeySelector.java:
```diff
@@ -0,0 +1,20 @@
+
+@Component
+public class RandomKeySelector {
+    private final ThreadLocalRandom random = ThreadLocalRandom.current();
```
다른 클래스처럼 개행 한 칸 띄워주세요~

drunkenhw replied:
```suggestion
    private static final ThreadLocalRandom RANDOM = ThreadLocalRandom.current();
```
으로 하면 안되나요?

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/service/ScrapperService.java:
```diff
@@ -27,11 +27,13 @@ public class ScrapperService {
 
     @Transactional
     public void scrap() {
+
```
필요하지 않은 개행은 없애주세요~

sosow0212 reviewed (changes requested):
컨벤션 위주로 리뷰 남겨봤습니다~

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/service/ScrapperService.java:
```diff
@@ -27,11 +27,13 @@ public class ScrapperService {
     @Transactional
     public void scrap() {
+
+        String SERVICE_KEY = randomKeySelector.getRandomKey();
```
```suggestion
        String serviceKey = randomKeySelector.getRandomKey();
```
상수가 아니니깐 이게 맞지 않을까요?

drunkenhw reviewed (changes requested):
궁금한거 위주로 리뷰남겼습니다~

be-student committed:
refactor: scrap 메서드를 리팩터링 한다 (#77)

* feat: mysql 파일들 git 관리 안 하도록 변경

* feat: yml 에 flyway 수정 및 sheduling 관리

* feat: 시작시 실행하도록 하는 기능을 추가한다

* fix: 외부 데이터가 잘못되어 깨지는 부분을 수정한다

* fix: test yml 에 필수 속성을 추가한다

* feat: 외부로 api 요청을 하는 기능을 추가한다

* feat: dto 작성

* feat: scrap 기능 추가

* feat: dto 에서 station 과 charger의 데이터 뽑아오는 기능을 추가한다

* feat: repository 와 도메인에 필요한 기능을 추가한다

* chore: 숫자를 상수로 분리한다

[#24]

* refactor: 안 쓰이는 코드 제거

[#24]

* feat: 빠진 equals 와 hashcode 수정

[#24]

* test: repository 테스트 작성

[#24]

* refactor: optional 대신 null 을 사용하도록 변경

[#24]

* refactor: 줄바꿈, 테스트 코드 메서드 네이밍 변경

[#24]

* refactor: 초 상수로 변경

[#24]

* refactor: 필요 없는 catch 제거

[#24]

* refactor: 안 쓰이는 코드 제거

[#24]

* refactor: requester 패키지 변경

[#24]

* refactor: 매직넘버 상수로 변경

[#24]

gabrielyoon7 committed:
feat: 서버로부터 수신 받은 충전소 데이터를 충전소 목록으로 보여주는 컴포넌트를 구현한다 (#82)

* rename: 폴더 이름을 googleMaps → google-maps 으로 변경

Co-authored-by: Dain Lee <feb-dain@users.noreply.github.com>
Co-authored-by: YoungWoo Kim <kyw0716@users.noreply.github.com>

[#75]

* refactor: 컴포넌트 폴더 구조 변경

Co-authored-by: Dain Lee <108778921+feb-dain@users.noreply.github.com>
Co-authored-by: kyw0716 <kyw0716@naver.com>

[#75]

* chore: CarFfeine을 고유명사로 등록

Co-authored-by: Dain Lee <feb-dain@users.noreply.github.com>
Co-authored-by: YoungWoo Kim <kyw0716@users.noreply.github.com>

[#75]

* refactor: 구글 지도 인스턴스를 전역상태로 관리하도록 수정

Co-authored-by: Dain Lee <108778921+feb-dain@users.noreply.github.com>
Co-authored-by: kyw0716 <kyw0716@naver.com>

[#75]

* feat: 디자인이 없는 임의의 컴포넌트에 수신받은 충전소를 렌더링하는 기능 구현

Co-authored-by: Dain Lee <108778921+feb-dain@users.noreply.github.com>
Co-authored-by: kyw0716 <kyw0716@naver.com>

[#75]

* refactor: google.maps.InfoWindow를 전역에서 관리하도록 개선

Co-authored-by: YoungWoo Kim <kyw0716@users.noreply.github.com>
Co-authored-by: Dain Lee <feb-dain@users.noreply.github.com>

[#75]

* refactor: 마커 인스턴스를 전역 상태로 관리

- 리렌더링 유발하지 않으면서 보장된 최신 상태를 받아오는 `getStoreSnapshot` 유틸 함수 추가

Co-authored-by: YoungWoo Kim <kyw0716@users.noreply.github.com>
Co-authored-by: Gabriel Ju Hyun, Yoon <gabrielyoon7@gmail.com>

[#75]

* feat: 충전소 목록 클릭시 해당 충전소 마커로 이동하고 간단 정보가 열리는 기능 구현

[#75]

---------

Co-authored-by: Dain Lee <feb.dain@gmail.com>
Co-authored-by: kyw0716 <kyw0716@naver.com>

sosow0212 committed:
feat: 테스트 격리를 위한 기능을 만든다 (#80)

* chore: rest-assured 종속성 추가

* feat: 통합 테스트용 어노테이션 추가 (테이블을 모두 Truncate + SpringBootTest(RANDOM_PORT) 속성을 가짐)

* refactor: 불필요한 static 제거

* refactor: IntegrationTest Extends해서 사용하도록 변경

kiarakim committed:
refactor: 상수와 변수의 이름 표기 방식 변경
